src/Ledger/Set/Theory.agda

```diff
@@ -126,3 +126,9 @@ indexedSum' f s = indexedSum ⦃ fromCommMonoid' it ⦄ f (s ᶠˢ)
 
 syntax indexedSumᵛ' (λ a → x) m = ∑[ a ← m ] x
 syntax indexedSum'  (λ a → x) m = ∑ˢ[ a ← m ] x
+
+module _ ⦃ _ : DecEq A ⦄ (P : A → Set) ⦃ _ : P ⁇¹ ⦄ {m : A ⇀ B} {X : ℙ A} where
+
+  ∈-filter-resᶜ-dom⁺ : ∀ {a} → P a × a ∉ X × ∃[ b ] (a , b) ∈ (m ˢ) → a ∈ filterˢ P (dom (m ∣ X ᶜ))
+  ∈-filter-resᶜ-dom⁺ (Pa , a∉X , bab∈m) = to ∈-filter (Pa , (∈-resᶜ-dom⁺ (a∉X , bab∈m)))
```

Comment:
I don't think this should be a separate lemma. This is essentially just a combination of the main properties of `filter` and restriction and there are `n^2` many such lemmas.

Reply:
I thought it would be useful to not have to think about how to prove this again in the future, and I thought this was the sort of thing James was suggesting in our last meeting... but maybe not.  Also, `n^2` many such lemmas? 🤔